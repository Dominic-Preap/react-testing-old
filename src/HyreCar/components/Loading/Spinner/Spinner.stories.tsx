import * as React from 'react';

import { storiesOf } from '@storybook/react';

import { Spinner } from './Spinner';

storiesOf('HyreCar|Components/Loading', module).add('Spinner', () => <Spinner />);
