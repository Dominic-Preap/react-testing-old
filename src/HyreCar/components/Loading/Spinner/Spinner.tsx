import * as React from 'react';

import { Theme } from '@material-ui/core/styles';
import { useTheme } from '@material-ui/styles';
import styled, { keyframes } from 'styled-components';

interface Props {
  color: string;
  grey: string;
}

const rotate = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const SpinnerContainer = styled.div`
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 100%;
`;

const SpinnerDiv = styled.div<Props>`
  animation: ${rotate} 0.5s linear infinite;
  border-radius: 100%;
  border-top-color: ${props => props.color} !important;
  border: 6px solid ${props => props.grey};
  height: 45px;
  width: 45px;
`;

export const Spinner = () => {
  const theme: Theme = useTheme();
  const props = {
    color: theme.palette.primary.main,
    grey: theme.palette.grey[200]
  };

  return (
    <SpinnerContainer>
      <SpinnerDiv {...props} />
    </SpinnerContainer>
  );
};
