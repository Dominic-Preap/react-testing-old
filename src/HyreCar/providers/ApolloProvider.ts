import ApolloClient from 'apollo-boost';

// https://github.com/apollographql/apollo-client/issues/3044

export const apolloClient = new ApolloClient({
  uri: 'https://countries.trevorblades.com',
  credentials: 'include',
  request: async operation => {
    operation.setContext({
      headers: { authorization: 'randomToken' }
    });
  }
});
