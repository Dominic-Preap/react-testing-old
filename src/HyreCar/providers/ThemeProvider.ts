import red from '@material-ui/core/colors/red';
import { createMuiTheme } from '@material-ui/core/styles';

// A custom theme for this app
export const theme = createMuiTheme({
  palette: {
    background: {
      default: '#f4f7fc'
    },
    error: {
      main: red.A400
    },
    primary: {
      main: '#009cec'
    },
    secondary: {
      main: '#19857b'
    },
    text: {
      primary: '#606770'
    }
  },
  typography: {
    fontFamily: ['Muli', 'Helvetica Neue', 'Helvetica', 'Arial', 'sans-serif'].join(',')
  }
});
