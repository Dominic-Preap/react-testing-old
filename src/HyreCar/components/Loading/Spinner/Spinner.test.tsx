import * as React from 'react';
import { create } from 'react-test-renderer';

import { ThemeProvider } from '@material-ui/styles';
import { theme } from 'HyreCar/providers/ThemeProvider';

import { Spinner } from './Spinner';

describe('Spinner component', () => {
  it('should render properly', () => {
    expect(
      create(
        <ThemeProvider theme={theme}>
          <Spinner />
        </ThemeProvider>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
