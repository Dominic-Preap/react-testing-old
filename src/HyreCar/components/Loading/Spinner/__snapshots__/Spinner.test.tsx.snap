// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should render properly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c1 {
  -webkit-animation: iECmZH 0.5s linear infinite;
  animation: iECmZH 0.5s linear infinite;
  border-radius: 100%;
  border-top-color: #009cec !important;
  border: 6px solid #eeeeee;
  height: 45px;
  width: 45px;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="#009cec"
  />
</div>
`;
